wgengine/filter/tailcfg.go

```diff
@@ -41,14 +43,15 @@ func MatchesFromFilterRules(pf []tailcfg.FilterRule) ([]Match, error) {
 		}
 
 		if len(r.IPProto) == 0 {
-			m.IPProto = append([]ipproto.Proto(nil), defaultProtos...)
+			m.IPProto = defaultProtosView
 		} else {
-			m.IPProto = make([]ipproto.Proto, 0, len(r.IPProto))
+			filtered := make([]ipproto.Proto, 0, len(r.IPProto))
```

Comment:
i guess we don't have a `views.Filter` or equivalent.

Reply:
Yeah, and we have slicex.Partition but it allocates a lot and doesn't support append. I'd like an AppendFilterFunc or something.